src/qt/bitcoingui.cpp

```diff
@@ -1038,49 +1038,51 @@ void BitcoinGUI::setNumBlocks(int count, const QDateTime& blockDate, double nVer
     progressBar->setToolTip(tooltip);
 }
 
-void BitcoinGUI::message(const QString &title, const QString &message, unsigned int style, bool *ret)
+void BitcoinGUI::message(const QString& title, QString message, unsigned int style, bool* ret)
 {
-    QString strTitle = tr("Bitcoin"); // default title
+    // Default title. On macOS, the window title is ignored (as required by the macOS Guidelines).
+    QString strTitle{PACKAGE_NAME};
     // Default to information icon
     int nMBoxIcon = QMessageBox::Information;
     int nNotifyIcon = Notificator::Information;
 
-    QString msgType;
+    bool prefix = !(style & CClientUIInterface::MSG_NOPREFIX);
+    style &= ~CClientUIInterface::MSG_NOPREFIX;
 
-    // Prefer supplied title over style based title
+    QString msgType;
     if (!title.isEmpty()) {
         msgType = title;
-    }
-    else {
+    } else {
         switch (style) {
         case CClientUIInterface::MSG_ERROR:
             msgType = tr("Error");
+            if (prefix) message = tr("Error: %1").arg(message);
```

Comment:
Why can't this use the existing translation of `tr("Error")`? Seems overkill to burden the translator with a format string.

Reply:
Please, see the https://github.com/bitcoin/bitcoin/pull/15894#issuecomment-487624823 by @ryanofsky.

Reply:
That explains that it can be done, not why.

Reply:
Maybe [this](https://github.com/bitcoin/bitcoin/blob/master/doc/translation_strings_policy.md#avoid-fragments) could explain that?
If not, would you mind providing the Bitcoin Core team's best practices for such cases?